name: 'orders-app' 
version: '1.0'
 
services: 
  mysql: 
    image: mysql:latest 
    container_name: mysql-server 
    restart: always
    env_file: 
      - ./env/mysql.env 
    networks: 
      - app-network 
    volumes: 
      - mysql-data:/var/lib/mysql 
      - ./db/init.sql:/docker-entrypoint-initdb.d/init.sql 
      - ./db/my.cnf:/etc/mysql/conf.d/my-custom.cnf:ro 
    healthcheck: 
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost", "-u", "novi", "--password=novi"] 
      interval: 10s 
      timeout: 10s 
      retries: 6 
 
  backend: 
    build: ./backend 
    container_name: backend 
    restart: always
    ports: 
      - 3000:3000 
    env_file: 
      - ./env/backend.env 
    networks: 
      - app-network 
    volumes: 
      - ./backend:/app 
      - /app/node_modules 
    depends_on: 
      mysql: 
        condition: service_healthy  
       
  frontend: 
    build: ./frontend 
    container_name: frontend 
    restart: always
    ports: 
      - "3001:3000" 
    networks: 
      - app-network 
    volumes:
      - ./frontend:/app 
      - /app/node_modules 
    depends_on: 
      - backend 
networks: 
  app-network: 
    driver: bridge 
volumes: 
  mysql-data:


